README.md

```diff
@@ -47,7 +47,7 @@ impl HelloService for HelloServer {
 
 fn main() {
     let server_handle = HelloServer.spawn("0.0.0.0:0").unwrap();
-    let client = hello_service::Client::new(server_handle.local_addr()).unwrap();
+    let client = hello_service::Client::new(server_handle.dialer()).unwrap();
```

Comment:
I'm fine making the example simpler. The handle should have two methods,
one returning Dialer and one creating a new client.
On Mar 2, 2016 8:47 PM, "shaladdle" <notifications@github.com> wrote:

> In README.md
> <https://github.com/google/tarpc/pull/27#discussion_r54835799>:
>
> > @@ -47,7 +47,7 @@ impl HelloService for HelloServer {
> >
> >  fn main() {
> >      let server_handle = HelloServer.spawn("0.0.0.0:0").unwrap();
> > -    let client = hello_service::Client::new(server_handle.local_addr()).unwrap();
> > +    let client = hello_service::Client::new(server_handle.dialer()).unwrap();
>
> My thoughts were more that the readme example should be the most common
> simple case. I think it's more common to have a client/server in different
> processes, rather than in the same one. So, I was thinking that using an
> address would be more familiar to some people.
>
> A separate issue is that we don't really have a full suite of examples..
> Just the benchmark. That should be addressed separately though.
>
> Returning client seems ok too. The one advantage of having a Dialer is
> that you can send bogus stuff over a stream, which is useful for verifying
> that we don't misbehave if someone sends junk to the server.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/tarpc/pull/27/files#r54835799>.
>